plugins/modules/source_control/gitlab/gitlab_user.py

```diff
@@ -107,6 +116,16 @@
       - Define external parameter for this user.
     type: bool
     default: no
+  provider:
```

Comment:
While looking at the code: why not call this `extern_provider`, so it's more obvious that this and `extern_uid` belong together?

Also from the code, it seems that the API allows a list of these. Would it make sense to allow a list of external logins here as well?

Reply:
> While looking at the code: why not call this extern_provider, so it's more obvious that this and extern_uid belong together?

It is named after the parameter used in the API call, which made more sense to us

> Also from the code, it seems that the API allows a list of these. Would it make sense to allow a list of external logins here as well?

When listing/getting users, external identities are shown as a list, but in user creation and modification it is added as in this module (ie with `provider` and `extern_uid`)
https://docs.gitlab.com/ee/api/users.html#user-creation
https://docs.gitlab.com/ee/api/users.html#user-modification

Reply:
> It is named after the parameter used in the API call, which made more sense to us

Since GitLab API parameter naming is not very consistent, I would prefer better naming in the module which makes it easier for users to see which parameters belong together.

> When listing/getting users, external identities are shown as a list, but in user creation and modification it is added as in this module

The API looks ... pretty basic here :) I'm not sure that it's a good idea to model the module after the API in this case. It looks to me that the API was somewhat designed under the assumption there is only one provider, and that assumption isn't true anymore. Ansible modules usually give users some more abstractions from such an API (or at least, they should IMO).

How about having a list of external identities instead, together with a boolean parameter whether to only add missing ones, or also remove ones that aren't present, and making the module properly manage that list for the user? I think that would be a much better UI for this module.

Reply:
(Not adding the removal functionality and that boolean parameter in this PR would be fine for me, that can be added later if someone wants to have that functionality, but allowing to add multiple ones is something I would really prefer.)

Reply:
I think it's fine to add it as a list, only thing I am worried about is when adding/changing multiple identities, you have to make multiple API calls, and when one goes wrong either half of the config is applied or we would have to add a rollback mechanism to revert the changes

Reply:
That's true, but that's a problem with every API that does not support transactions and does not have "magic" calls which happen to do everything you want to do in one step. I think it's acceptable if only parts of the changes are applied (I would do removing of login providers as the last step to avoid the situation that all have been removed and adding a new one fails).

Reply:
Changes have been added, you can now add multiple identities at once and overwrite them with a boolean